get_ecg.py

```diff
@@ -0,0 +1,173 @@
+            self.brady_threshold = brady_threshold
+            self.tachy_threshold = tachy_threshold
+
+        self.raw_bunches = []
+        self.total_peaks = []
+
+    def get_max_peak(self):
+        """
+           Function that will find local maximum of an array i.e peaks
+
+              :return total_peaks: list of all peaks detected, arranged
+               into chunks based on update_time
+              :rtype: list
+
+           """
+        min_dist = 150
+        if isinstance(self.update_time, float):
+            self.update_time = int(self.update_time)
+        total_time = self.time_array[-1] - self.time_array[1]
+        bunches = total_time/self.update_time
+
+        if total_time / self.update_time < 1:
+            raise ValueError("Update time is longer than signal length")
+        divided_voltage_array = np.array_split(self.voltage_array, bunches)
+        divided_time_array = np.array_split(self.time_array, bunches)
+        length = len(self.voltage_array)
+
+        if length % bunches != 0:
+            np.delete(divided_voltage_array, -1)
+        for i in range(len(divided_voltage_array)):
+            dump = []
+            new_time_array = divided_time_array[i]
+            new_voltage_array = divided_voltage_array[i]
+            max_peaks = []
+
+            mx, mn = -np.Inf, np.Inf
+
+            for index, (x, y) in enumerate(zip(new_time_array,
+                                               new_voltage_array)):
+                if y > mx:
+                    mx = y
+                    mxpos = x
+                if y < mn:
+                    mn = y
+
+                # Looking for max
+                if y < mx and mx != np.Inf:
+                    # Maxima peak candidate found
+                    if new_voltage_array[index:index + min_dist].max() < mx:
+                        max_peaks.append([mxpos, mx])
+                        dump.append(True)
+                        # set algorithm to only find minima now
+                        mn = np.Inf
+                        mx = np.Inf
+                        if index + min_dist >= length:
+                            # end is within lookahead no more peaks
+                            break
+                        continue
+
+                if y > mn and mn != -np.Inf:
+                    if new_voltage_array[index:index + min_dist].min() > mn:
+                        dump.append(False)
+                        mn = -np.Inf
+                        mx = -np.Inf
+                        if index + min_dist >= length:
+                            break
+
+            self.total_peaks.append(max_peaks)
+            # Remove the false hit on the first value of the y_axis
+            try:
+                if dump[0]:
+                    max_peaks.pop(0)
+                del dump
+            except IndexError:
+                    # no peaks were found
+                    print("No peaks were found")
+
+        return self.total_peaks
```

Comment:
You don't have to return an object attribute.

Reply:
IMHO In this case, I think it may make sense to have this function `get_max_peak[s]` return a value to the caller based on the fact that the caller probably wants that information right back if they're executing this function? It is certainly odd to return an attribute like this because usually there's an internal variable holding the info needed to return